### PR TITLE
vitest: add basic docs

### PR DESCRIPTION
**Pull request by gcanti**

### CODE REVIEW

File: packages/vitest/README.md [hunk 1]
```diff
@@ -1,64 +1,230 @@
-# Effect
+# Introduction
 
-Welcome to Effect, a powerful TypeScript framework that provides a fully-fledged functional effect system with a rich standard library.
+Welcome to your guide on testing Effect applications using `vitest` and the `@effect/vitest` package. This guide is designed to help beginners efficiently set up their testing environment and create robust, maintainable tests. By integrating `vitest` with the Effect library, you enhance your testing capabilities, allowing for more expressive tests. Let’s start by setting up the necessary tools and then dive into crafting effective test cases.
```
Comment (IMax153) [inline packages/vitest/README.md:3]: This sounds a bit GPT-ish 😅 - maybe the following?

```suggestion
Welcome to your guide on testing Effect applications using `vitest` and the `@effect/vitest` package! `@effect/vitest` is designed to help simplify running Effect-based tests through `vitest`. 

In the guide below, we will first start by setting up the required dependencies. Then, we will dive into a few examples of how to use `@effect/vitest` to create some Effect-based test cases.
```

File: packages/vitest/README.md [hunk 2]
```diff
@@ -1,64 +1,230 @@
...
+pnpm add -D vitest
 ```
-{
-  // ...
-  "compilerOptions": {
-    // ...
-    "strict": true,
-  }
+
+Next, install the `@effect/vitest` package. This package integrates the Effect framework with Vitest for enhanced testing capabilities.
```
Comment (IMax153) [inline packages/vitest/README.md:13]:
```suggestion
Next, install the `@effect/vitest` package which facilitates running Effect-based tests through `vitest`.
```

File: packages/vitest/README.md [hunk 3]
```diff
@@ -1,64 +1,230 @@
...
+
+Here's how to use `it.effect` to write your tests:
+
+```ts
+import { it } from "@effect/vitest"
+
+it.effect("test name", () => EffectContainingAssertions, timeout: number | TestOptions = 5_000)
+```
+
+When using `it.effect`, the `TestContext` is automatically injected, simplifying test setup and execution.
```
Comment (IMax153) [inline packages/vitest/README.md:47]:
```suggestion
When using `it.effect`, the `TestContext` is automatically injected, which allows tests to have access to services designed to facilitate testing (such as the [`TestClock`](#using-the-testclock)).
```

File: packages/vitest/README.md [hunk 4]
```diff
@@ -1,64 +1,230 @@
...
+
+## Testing Successful Operations
+
+To demonstrate, let's create a test for a function that divides two numbers but fails if the divisor is zero:
```
Comment (IMax153) [inline packages/vitest/README.md:51]:
```suggestion
Let's create a test for a function that divides two numbers but fails if the divisor is zero:
```

File: packages/vitest/README.md [hunk 5]
```diff
@@ -1,64 +1,230 @@
...
+    const result = yield* divide(4, 0).pipe(Effect.exit)
+    expect(result).toStrictEqual(Exit.fail("Cannot divide by zero"))
+  })
+)
+```
+
+## Using the TestClock
 
-[![Introduction to Effect](https://img.youtube.com/vi/SloZE4i4Zfk/maxresdefault.jpg)](https://youtu.be/SloZE4i4Zfk)
+The `TestClock` is seamlessly integrated into your tests when you use `it.effect`, making it easy to simulate the passage of time. If you want to execute your tests in the real, live Effect environment instead, you can use `it.live`.
```
Comment (IMax153) [inline packages/vitest/README.md:102]:
```suggestion
When using `it.effect`, a `TestContext` is provided to your program which provides access to several services designed to facilitate testing. One such service is the `[TestClock`](https://effect.website/docs/guides/testing/testclock) which is designed to simulate the passage of time. 

**Note**: To utilize the default, non-testing services in your tests you can use `it.live`.
```

**Review by IMax153 — CHANGES_REQUESTED** [review]

File: packages/vitest/README.md [hunk 6]
```diff
@@ -1,64 +1,234 @@
...
+
+it.effect("run the test with the test environment", () =>
+  Effect.gen(function* () {
+    yield* logNow // prints 0
+  })
+)
+
+it.effect("run the test with the test environment and the time adjusted", () =>
+  Effect.gen(function* () {
+    yield* TestQlock.adjust("1000 millis")
```
Comment (vecerek) [inline packages/vitest/README.md:139]:
```suggestion
    yield* TestClock.adjust("1000 millis")
```

**Review by IMax153 — APPROVED** [review]
LGTM 👍 Thanks @gcanti !

**Pull request merged**